src/mlpack/methods/ann/rnn_impl.hpp

```diff
@@ -633,7 +633,8 @@ void RNN<
     MatType
 >::Shuffle()
 {
-  ShuffleData(predictors, responses, predictors, responses);
+  ShuffleData(predictors, responses, sequenceLengths,
+      predictors, responses, sequenceLengths);
```

Comment:
I know that the choice to have `sequenceLengths` ignored if it's empty makes the code here easier, but it doesn't match the behavior of the other `ShuffleData()` overloads; would you be opposed to making `ShuffleData()` expect the weights/lengths to be the same length as the data, and then just using an `if` here to determine which overload to call (e.g. with lengths or without them)?  My motivation is avoiding behavior that doesn't match the documentation.